appender/elasticsearch/src/test/java/org/apache/karaf/decanter/appender/elasticsearch/TestElasticsearchAppender.java

```diff
@@ -89,12 +89,15 @@ public void test() throws Exception {
 
         HttpHost host = new HttpHost(HOST, HTTP_PORT, "http");
         RestClient client = RestClient.builder(new HttpHost[]{ host }).build();
-        Response response = client.performRequest("GET", "/_count", Collections.EMPTY_MAP);
-
-        String responseString = EntityUtils.toString(response.getEntity());
 
+        Date start = new Date();
+        long timeout = 5000L;
+        String responseString = "";
+        while (!responseString.contains("\"count\":3") || (new Date().getTime() - start.getTime() < timeout)) {
```

Comment:
1. It's pretty agressive for the CPU ;) I would do a little sleep in the loop
2. You don't have to deal with the timeout yourself, you can just put on junit level.

Basically, you can just do:

```
while (!responseString.contains("\"count\":3")) {
  Thread.sleep(200);
  Response response = client.performRequest("GET", "/_count", Collections.EMPTY_MAP);
  responseString = EntityUtils.toString(response.getEntity());
}
```
and define the timeout like this:
```
@Test(timeout = 60L)
public void test() throws Exception {
  ...
}
```

Reply:
@jbonofre you're right :)
I pushed the changes.
I removed the Assert test because it will always be true (condition of the loop).